### PR TITLE
Be inclusive of which files to bundle with gems

### PR DESCRIPTION
**Pull request by vishrutshah**
This change is about being inclusive on which files to be bundled with gems instead of rejecting files do not required from git ls-files -z

This way we will not unintentional bundle files like Gemfile, *.gemspec, Rakefile etc... like we already did that until recently.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@vishrutshah__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: azure_mgmt_authorization/azure_mgmt_authorization.gemspec [hunk 1]
```diff
@@ -18,7 +18,7 @@ Gem::Specification.new do |spec|
   spec.homepage      = 'https://aka.ms/azure-sdk-for-ruby'
   spec.license       = 'MIT'
 
-  spec.files         = `git ls-files -z`.split("\x0").reject { |f| f.match(%r{^(test|spec|features)/}) }
+  spec.files         = Dir["LICENSE.txt", "lib/**/*"]
```
Comment (veronicagg) [inline azure_mgmt_authorization/azure_mgmt_authorization.gemspec]: do we want to exclude Rakefile and gemspec file?

Comment (vishrutshah) [inline azure_mgmt_authorization/azure_mgmt_authorization.gemspec]: Yup. I believe that they are not required.

**Comment by veronicagg** [comment]
@vishrutshah LTMG!

**Pull request merged**